fossilize_db.cpp

```diff
@@ -610,27 +610,53 @@ DatabaseInterface *create_zip_archive_database(const char *path, DatabaseMode mo
 	return db;
 }
 
+/* Fossilize StreamArchive database format version 6:
+ *
+ * The file consists of a header, followed by an unlimited series of "entries".
+ *
+ * All multi-byte entities are little-endian.
+ *
+ * The file header is as follows:
+ *
+ * Field           Type           Description
+ * -----           ----           -----------
+ * magic_number    uint8_t[12]    Constant value: "\x81""FOSSILIZEDB"
+ * unused1         uint8_t        Currently unused. Must be zero.
+ * unused2         uint8_t        Currently unused. Must be zero.
+ * unused3         uint8_t        Currently unused. Must be zero.
+ * version         uint8_t        StreamArchive version: 6
+ *
+ *
+ * Each entry follows this format:
+ *
+ * Field           Type                              Description
+ * -----           ----                              -----------
+ * tag             unsigned char[40 - hash_bytes]    Application-defined 'tag' which groups entry types. Stored as hexadecimal big-endian ASCII.
+ * hash            unsigned char[hash_bytes]         Application-defined 'hash' to identify this entry. Stored as hexadecimal big-endian ASCII.
```

Comment:
Endianness of the hash is technically application defined. It should be treated as 40 opaque hex values.